base/main_test_bucket_pool.go

```diff
@@ -400,11 +400,16 @@ func (tbp *TestBucketPool) printStats() {
 	tbp.Logf(ctx, "==========================")
 
 	tbp.unclosedBucketsLock.Lock()
+	unclosedBucketWarnings := ""
 	for testName, buckets := range tbp.unclosedBuckets {
 		for bucketName := range buckets {
 			tbp.Logf(ctx, "WARNING: %s left %s bucket unclosed!", testName, bucketName)
+			unclosedBucketWarnings += fmt.Sprintf("%s left %s bucket unclosed!\n", testName, bucketName)
 		}
 	}
+	if unclosedBucketWarnings != "" {
+		panic(unclosedBucketWarnings)
```

Comment:
Originally, this was put in when a Leaky TestBucket was to be passed in to the RT in-order to make sure the bucket was closed manually at the end of the test and can't be missed. I left this change in because it would be useful in general as it is quite easy to miss a bucket being left unclosed when not going through all the test logs.